src/equations_set_routines.F90

```diff
@@ -1792,22 +1795,59 @@ SUBROUTINE EQUATIONS_SET_BACKSUBSTITUTE(equationsSet,BOUNDARY_CONDITIONS,err,err
+                                                      eqColLocalIdx = &
+                                                        & COLUMN_DOMAIN_MAPPING%GLOBAL_TO_LOCAL_MAP &
+                                                        & (equations_column_number)%LOCAL_NUMBER(domain_idx)
+                                                      ! We do not consider ghosts
+                                                      IF (eqColLocalIdx <= &
+                                                        & COLUMN_DOMAIN_MAPPING%TOTAL_NUMBER_OF_LOCAL) THEN
+                                                        ! Local dof on rhs
+                                                        variable_dof=eqColLocalIdx!equations_column_number
+                                                        ! original line: delete!
+                                                        ! variable_dof = equations_column_idx-ROW_INDICES(equations_row_number)+1
+                                                         ! Global dof in matrix
+                                                        MATRIX_VALUE=equationsMatrixData(equations_column_idx)
+                                                        DEPENDENT_VALUE=DEPENDENT_PARAMETERS(variable_dof)
+                                                        RHS_VALUE=RHS_VALUE+MATRIX_VALUE*DEPENDENT_VALUE
+                                                      END IF
+                                                    END IF
+                                                  END DO
                                                 ENDDO !equations_column_idx
+
+                                                !CASE block storage (=FULL) above
+                                                !Back substitute
+                                                !Loop over the local columns of the equations matrix
+                                                !DO equations_column_idx=1,COLUMN_DOMAIN_MAPPING%TOTAL_NUMBER_OF_LOCAL
+                                                !  equations_column_number=COLUMN_DOMAIN_MAPPING%LOCAL_TO_GLOBAL_MAP( &
+                                                !    & equations_column_idx)
+                                                !  variable_dof=equations_column_idx
+                                                !  MATRIX_VALUE=equationsMatrixData(equations_row_number+ &
+                                                !    & (equations_column_number-1)*vectorMatrices%totalNumberOfRows)
+                                                !  DEPENDENT_VALUE=DEPENDENT_PARAMETERS(variable_dof)
+                                                !  RHS_VALUE=RHS_VALUE+MATRIX_VALUE*DEPENDENT_VALUE
+                                                !ENDDO !equations_column_idx
+
                                               CASE(BOUNDARY_CONDITION_DOF_FIXED)
                                                 !Do nothing
```

Comment:
Do we leave RHS_VALUE as 0.0 here if BOUNDARY_CONDITION_DOF_FIXED? We will set rhs_variable_dof to 0.0 then?